### PR TITLE
Update Task objects and add shortcut to cancel tasks

### PR DESCRIPTION
**Pull request by jsh2134**
add task.cancel() and also makes Task undateable so that priority can be set

### CODE REVIEW

File: solvebio/resource/task.py [hunk 1]
```diff
@@ -27,3 +27,15 @@ def child_object(self):
     def follow(self):
         """ Follow the child object but do not loop """
         self.child_object.follow(loop=False)
+
+    def cancel(self):
```
Comment (jsh2134) [inline solvebio/resource/task.py]: might actually be cleaner to make Task and UpdateableResource and send the update there.

File: solvebio/resource/task.py [hunk 2]
```diff
@@ -27,3 +27,15 @@ def child_object(self):
...
+        """ Cancel a task """
+        if 'cancel' not in self.available_actions:
```
Comment (davecap) [inline solvebio/resource/task.py]: The API already validates this and provides a message about the current status as well.

File: solvebio/resource/task.py [hunk 3]
```diff
@@ -27,3 +27,15 @@ def child_object(self):
...
+            # It may not be queued/running or another
+            # possible reason.
+            print("Task can not be canceled.")
+            return
+
+        child_ = self.child_object
```
Comment (davecap) [inline solvebio/resource/task.py]: Does it have to be the child object? Can't we just set cancel on the task object itself?

Comment (davecap) [reply solvebio/resource/task.py]: Or is this what you were referring to with your comment above?

**Comment by davecap** [comment]
Add UpdateableAPIResource class to the the Task resource object.

**Pull request merged**